modules/sdk/src/main/scala/org/tessellation/sdk/infrastructure/consensus/ConsensusManager.scala

```diff
@@ -177,6 +190,28 @@ object ConsensusManager {
             .handleErrorWith(logger.error(_)(s"Error checking for consensus state update {key=${key.show}}"))
         }.void
 
+      private def getPeerCandidates(peers: List[Peer]): F[List[Peer]] = {
+        case class PeerSnapshot(ordinal: SnapshotOrdinal, hash: Hash, peer: Peer)
+        val groupFn: PeerSnapshot => (SnapshotOrdinal, Hash) = v => (v.ordinal, v.hash)
+
+        for {
+          peerSnaps: List[PeerSnapshot] <- peers.toList.traverse(peer =>
+            for {
+              signedSnapshot <- gsClient.getLatest(peer)
```

Comment:
`getLatest` as the name suggests gets the outcome of the latest consensus, what's the point of comparing hashes of different snapshots?

Reply:
The idea is to determine the majority branch. If there are `n` `Peer`s where `n > 1`, not all the `Peer`s may agree on the latest snapshot (i.e. they are on different forks). We compare the hash and ordinals to determine what the majority branch is so that we can down against one of those `Peer`s.

Reply:
There's no logic here to handle the case of timing differences with respect to ordinal, but I think Chuong has made another ticket to capture this. I.e. if ordinal 1000 and 1001 are returned by different queries (but are otherwise on the same fork), for now we're ignoring that and just capturing majority of both.

Reply:
> The idea is to determine the majority branch. If there are n Peers where n > 1, not all the Peers may agree on the latest snapshot (i.e. they are on different forks).

Different case is when there are ordinals at the same ordinal number but with different hashes (nodes are misaligned and indeed in different forks) and different case is when you compare hash of snapshot at ordinal 10 and hash of snapshot at ordinal 11 - they will be different even if both nodes are in the same fork and are aligned.

@ryle-goehausen I see no point in introducing issues or partially working code and fixing it later. (exception for this is when we merge it to a feature branch or agree to somehow delay deployments on mainnet)

Reply:
I agree with @marcinwadon . Submitting a broken functionality only to create a technical debt, which needs to be addressed later on is a bad idea.  Please fix it, comparing hashes of snapshots with different ordinals is simply pointless.

Reply:
We have a separate ticket for @cngo-github to implement the process of determining latest ordinal. Unless I am mistaken this PR is strictly an improvement upon the prior implementation (which might be referred to as technical debt that already exists, or a bug that is already in the code which this is not making worse.) This should be an incremental improvement upon the existing functionality by at least introducing some check of multiple peers (even if not accounting for ordinals.) Since we already have a ticket to fix the ordinal and you're not okay with splitting up these PRs into separate items, I'll discuss with Chuong the next steps to fixing the ordinal query as part of this PR.

Reply:
I will implement the process of determining the latest ordinal as part of this PR.